python_modules/libraries/dagster-aws/dagster_aws/pipes/message_readers.py

```diff
@@ -115,21 +115,37 @@ class CloudWatchEvent(TypedDict):
 class PipesCloudWatchMessageReader(PipesMessageReader):
     """Message reader that consumes AWS CloudWatch logs to read pipes messages."""
 
-    def __init__(self, client: Optional[boto3.client] = None):  # pyright: ignore (reportGeneralTypeIssues)
+    def __init__(
+        self,
+        client=None,
+    ):
         """Args:
         client (boto3.client): boto3 CloudWatch client.
         """
         self.client = client or boto3.client("logs")
 
+    @property
+    def read_location_known(self) -> bool:
+        return False
+
+    @contextmanager
+    def get_writer_params(self) -> Iterator[PipesParams]:
+        yield {PipesDefaultMessageWriter.STDIO_KEY: PipesDefaultMessageWriter.STDERR}
+
     @contextmanager
     def read_messages(
         self,
         handler: PipesMessageHandler,
+        extra_params: Optional[PipesParams] = None,
     ) -> Iterator[PipesParams]:
         self._handler = handler
+
+        assert extra_params is not None, "extra_params must be provided."
```

Comment:
we typically use check.invariant to check business logic like this as `assert` can be compiled out by a user